### PR TITLE
plugin/rest: use proper settings for http.Transport

### PR DESCRIPTION
**Pull request by srenatus**
[Please see the comments here](https://github.com/open-policy-agent/opa/compare/master...srenatus:sr/bugfix/default-transport#diff-f4076aa50c92ccd52544c8f6e44238e6R165); the fix in `server_test.go` is in the same vein, but less important 😃

### CODE REVIEW

File: plugins/rest/rest.go [hunk 1]
```diff
@@ -159,11 +162,35 @@ func New(config []byte, opts ...func(*Client)) (Client, error) {
 		return Client{}, err
 	}
 
+	// Ensure we use a http.Transport with proper settings: the zero values are not
```
Comment (tsandall) [inline plugins/rest/rest.go]: Could you adjust this comment slightly to specify _why_ the default settings are not a good choice? I looked at the linked issue and I _think_ the answer is that it leads to leaks but I'm not sure.

**Comment by tsandall** [comment]
@pwmorreale I'm also a huge fan of good commit messages :heart_eyes_cat:

FWIW, I think that @srenatus also shares this value. For example: https://github.com/open-policy-agent/opa/commit/59f7020d2d9efd627c61b511800674dc3cbe8a88.

Also, we actually have this documented as part o the CONTRIBUTING guide for OPA: https://github.com/open-policy-agent/opa/blob/master/CONTRIBUTING.md#commit-messages

**Comment by srenatus** [comment]
@pwmorreale thanks for the reminder -- have to admit, a _2-minute PR_ seldom is a good one 😅 I'll brush this up with a little more time today.

**Comment by tsandall** [comment]
LGTM..

**Pull request merged**